src/containers/tutor-home-page/general-info-step/GeneralInfoStep.tsx

```diff
@@ -21,32 +21,67 @@ import { useStepContext } from '~/context/step-context'
 import { validations } from '~/components/user-steps-wrapper/constants'
 import { styles } from '~/containers/tutor-home-page/general-info-step/GeneralInfoStep.styles'
 import { defaultResponses } from '~/constants'
+import { useAppSelector } from '~/hooks/use-redux'
+import { Country, UserGeneralInfo, UserRole } from '~/types'
+
+interface GeneralInfoStepProps {
+  btnsBox: ReactNode
+  isUserFetched: boolean
+  setIsUserFetched: (isUserFetched: boolean) => void
+}
+
+type UserName = { firstName: string; lastName: string }
 
-const GeneralInfoStep = ({ btnsBox, isUserFetched, setIsUserFetched }) => {
+const GeneralInfoStep = ({
+  btnsBox,
+  isUserFetched,
+  setIsUserFetched
+}: GeneralInfoStepProps) => {
   const { t } = useTranslation()
   const { isLaptopAndAbove, isMobile } = useBreakpoints()
   const { stepData, handleGeneralInfo } = useStepContext()
-  const { userId, userRole } = useSelector((state) => state.appMain)
+  const { userId, userRole } = useAppSelector((state) => state.appMain)
   const generalInfo = stepData.generalInfo
 
+  const getCountries = useCallback(() => LocationService.getCountries(), [])
+
+  const {
+    loading: loadingCountries,
+    response: countries,
+    fetchData: fetchCountries
+  } = useAxios({
+    service: getCountries,
+    fetchOnMount: false,
+    defaultResponse: defaultResponses.array as Country[]
+  })
+
   const {
     handleInputChange,
     handleBlur,
     handleNonInputValueChange,
     data,
     errors
-  } = useForm({
+  } = useForm<UserGeneralInfo>({
     initialValues: generalInfo.data,
-    initialErrors: generalInfo.errors,
-    validations
+    initialErrors: {
+      city: generalInfo.errors['city'] || '',
+      country: generalInfo.errors['country'] || '',
+      firstName: generalInfo.errors['firstName'] || '',
+      lastName: generalInfo.errors['lastName'] || '',
+      professionalSummary: generalInfo.errors['professionalSummaryy'] || ''
+    },
+    ...validations
   })
 
-  const filterOptions = (options, state) => {
+  const filterOptions = (
+    options: unknown[],
```

Comment:
maybe, its better to use here options: string[] to be more specific with the type we expect? or please tell why you decided to use unknown[] here.